### PR TITLE
ask-password: give a hint to cancel echo

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/test/test-ask-password-api.c [hunk 1]
```diff
@@ -10,10 +10,13 @@ static void ask_password(void) {
         _cleanup_strv_free_ char **ret = NULL;
 
         r = ask_password_tty(-1, "hello?", "da key", 0, 0, NULL, &ret);
-        assert(r >= 0);
-        assert(strv_length(ret) == 1);
-
-        log_info("Got %s", *ret);
+        if (r == -ECANCELED)
+                assert_se(ret == NULL);
```
Comment (yuwata) [inline src/test/test-ask-password-api.c:13]: Why `assert_se()` instead of `assert()` here?

File: src/test/test-ask-password-api.c [hunk 2]
```diff
@@ -10,10 +10,13 @@ static void ask_password(void) {
...
+        else {
+                assert(r >= 0);
+                assert(strv_length(ret) == 1);
+                log_info("Got \"%s\"", *ret);
```
Comment (yuwata) [inline src/test/test-ask-password-api.c:17]: Is it safe to log this information?

File: src/test/test-ask-password-api.c (hunk 1 above)
Comment (keszybz) [reply src/test/test-ask-password-api.c:13]: Because we're in test here.

File: src/test/test-ask-password-api.c (hunk 2 above)
Comment (keszybz) [reply src/test/test-ask-password-api.c:17]: Yeah, that's just for testing.

File: src/test/test-ask-password-api.c (hunk 1 above)
Comment (yuwata) [reply src/test/test-ask-password-api.c:13]: Ouch. Sorry for the noise.

File: src/test/test-ask-password-api.c (hunk 2 above)
Comment (keszybz) [reply src/test/test-ask-password-api.c:17]: But there should be assert_se here too. I pushed an updated version.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**